src/librustc_typeck/check/mod.rs

```diff
@@ -1771,16 +1775,37 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     fn default_type_parameters(&self) {
         use middle::ty::error::UnconstrainedNumeric::Neither;
         use middle::ty::error::UnconstrainedNumeric::{UnconstrainedInt, UnconstrainedFloat};
+
+        // Defaulting inference variables becomes very dubious if we have
+        // encountered type-checking errors. Therefore, if we think we saw
+        // some errors in this function, just resolve all uninstanted type
+        // varibles to TyError.
+        if self.infcx().is_tainted_by_errors() {
```

Comment:
@arielb1 

> If you are doing your best attempt to break out of inference, why not just do it?

You mean why unify with error? The idea was to taint these variables in a way that would cause other errors to be suppressed if they should refer to them. Annoyingly, it didn't entirely work because of integral variables, so I still needed a few tests elsewhere.